InnerEye/ML/configs/classification/CovidHierarchicalModel.py

```diff
@@ -0,0 +1,241 @@
+    def get_image_sample_transforms(self) -> ModelTransformsPerExecutionMode:
+        config = load_ssl_augmentation_config(path_linear_head_augmentation_cxr)
+        train_transforms = ScalarItemAugmentation(
+            Compose(
+                [DicomPreparation(), create_chest_xray_transform(config, apply_augmentations=True)]))
+        val_transforms = ScalarItemAugmentation(
+            Compose(
+                [DicomPreparation(), create_chest_xray_transform(config, apply_augmentations=False)]))
+
+        return ModelTransformsPerExecutionMode(train=train_transforms,
+                                               val=val_transforms,
+                                               test=val_transforms)
+
+    def create_model(self) -> LightningModule:
+        """
+        This method must create the actual Lightning model that will be trained.
+        """
+        if self.use_pretrained_model:
+            path_to_checkpoint = self._get_ssl_checkpoint_path()
+
+            model = create_ssl_image_classifier(
+                num_classes=self.num_classes,
+                pl_checkpoint_path=str(path_to_checkpoint),
+                freeze_encoder=self.freeze_encoder)
+
+        else:
+            encoder = create_ssl_encoder(encoder_name=EncoderName.densenet121.value)
+            model = SSLClassifier(num_classes=self.num_classes,
+                                  encoder=encoder,
+                                  freeze_encoder=self.freeze_encoder,
+                                  class_weights=None)
+        # Next args are just here because we are using this model within an InnerEyeContainer
+        model.imaging_feature_type = ImagingFeatureType.Image  # type: ignore
+        model.num_non_image_features = 0  # type: ignore
+        model.encode_channels_jointly = True  # type: ignore
+        return model
+
+    def _get_ssl_checkpoint_path(self) -> Path:
+        # Get the SSL weights from the AML run provided via "pretraining_run_recovery_id" command line argument.
+        # Accessible via extra_downloaded_run_id field of the config.
+        assert self.extra_downloaded_run_id is not None
+        assert isinstance(self.extra_downloaded_run_id, RunRecovery)
+        ssl_path = self.checkpoint_folder / "ssl_checkpoint.ckpt"
+        logging.info(f"name_of_checkpoint: {self.name_of_checkpoint}")
+        if not ssl_path.exists():  # for test (when it is already present) we don't need to redo this.
+            if self.name_of_checkpoint is not None:
+                path_to_checkpoint = self.extra_downloaded_run_id.checkpoints_roots[0] / self.name_of_checkpoint
+            else:
+                try:
+                    path_to_checkpoint = self.extra_downloaded_run_id.get_best_checkpoint_paths()[0]
+                    assert path_to_checkpoint.exists()
+                except AssertionError:
+                    path_to_checkpoint = self.extra_downloaded_run_id.get_recovery_checkpoint_paths()[0]
```

Comment:
not sure it's a good idea to use assertions for control flow, as they can be disabled (the one below is fine, as it's just verifying the status). any special reason not to just use a plain `if`?

Reply:
Yes good point